src/features/book/bookSlice.js

```diff
@@ -0,0 +1,54 @@
+import { createSlice, createAsyncThunk } from '@reduxjs/toolkit';
+import axios from 'axios';
+
+const url = 'https://us-central1-bookstore-api-e63c8.cloudfunctions.net/bookstoreApi/apps/WA5eot1mOTs5Y1g1bAoc/books';
+
+const initialState = {
+  booksByItem: {},
+  isLoading: true,
+  error: null,
+};
+
+export const getBookItems = createAsyncThunk(
+  'book/getBookItems',
+  async (thunkAPI) => {
+    try {
+      const resp = await axios(url);
+      return resp.data;
+    } catch (error) {
+      return thunkAPI.rejectWithValue('something went wrong');
+    }
+  },
+);
+
+const bookSlice = createSlice({
+  name: 'Bookstore',
+  initialState,
+  reducers: {
+    addBooks: (state, action) => {
+      const { itemId, books } = action.payload;
+      state.booksByItem[itemId] = books;
+    },
+    removeBook: (state, action) => {
+      const { itemId, bookId } = action.payload;
+      state.booksByItem[itemId] = state.booksByItem[itemId].filter((book) => book.id !== bookId);
+    },
+  },
```

Comment:
- [ ] Great job on your progress! 😄 However, I'd like to kindly remind you about an important project requirement. According to the project specifications, it's necessary to refactor the `addBooks` and  `removeBook` reducers in a way that returns a promise and adds and removes a new book via an API call.

  - To meet this requirement, I recommend considering the implementation of `createAsyncThunk`. This will allow you to create an action that can make a request to your API and seamlessly add and remove a new book.

Happy coding! 😊🚀🍀🙌